util/testsuite/testsuite.go

```diff
@@ -291,7 +297,7 @@ func (ts *TestSuite) GenerateTestCertificate(blockHash hash.Hash) *block.Certifi
 	_, priv3 := ts.RandomBLSKeyPair()
 	_, priv4 := ts.RandomBLSKeyPair()
 
-	sigs := []*bls.Signature{
+	sigs := []*bls.Signature{ //nolint
```

Comment:
Why?

Reply:
it makes tests more heavy and complex.

Reply:
Hopefully we can use generics here. @amirvalhalla any update?

Reply:
okay, I will wait for updates from him.
also, I can start work on this if you wish.

Reply:
Yes, You may try as well

Reply:
@b00f 
@kehiy 

I will work on make generic these methods until end of this weekend , please be patient :)

Reply:
@b00f 
@kehiy 

Hi guys
Unfortunately, right now we can't use generics for `PrivateKey`, `PublicKey`,`Signature` etc... the reason is if we want to use generics we should change the structure of a lot of functions and interfaces to support generics.

e.g. in the `bls` package we implemented `PrivateKey` interface methods so if we want to use generics for `EqualsTo`  we should convert `PrivateKey` struct to `PrivateKey[T any]` this change makes some conflicts for other methods  like `PrivateKeyFromString` method, another reason is `Crypto.PrivateKey` interface is not generic so if we also change it to generic like `Crypto.PrivateKey[T any]` another issue is we can't cast it to another type.(if want to do that we should use embedded struct).

if generics are necessary which I think we should consider it, we should create another issue and discuss about what methods or packages should be generic then refactor our codes.